src/SassBuilder.php

```diff
@@ -26,24 +26,26 @@ public function __construct(
         private readonly string $projectRootDir,
         private readonly ?string $binaryPath,
         private readonly bool $embedSourcemap,
+        private readonly ?array $sassOptions = [],
     ) {
     }
 
     public function runBuild(bool $watch): Process
     {
         $binary = $this->createBinary();
 
-        $args = $this->getScssCssTargets();
+        $options = new SassOptions($this->sassOptions);
+        if ($this->embedSourcemap) {
+            $options->embedSourceMap();
+        }
 
         if ($watch) {
             $args[] = '--watch';
         }
 
-        if ($this->embedSourcemap) {
-            $args[] = '--embed-source-map';
-        }
+        $args = $this->getScssCssTargets();
```

Comment:
Moving this after the handling of `$watch` that was adding things in the array is broken.